lib/beaker/dsl/install_utils/puppet_utils.rb

```diff
@@ -197,7 +197,7 @@ def install_puppet(opts = {})
 
             # Certain install paths may not create the config dirs/files needed
             on host, "mkdir -p #{host['puppetpath']}" unless host[:type] =~ /aio/
-            on host, "echo '' >> #{host.puppet['hiera_config']}"
+            #on host, "echo '' >> #{host.puppet['hiera_config']}"
```

Comment:
This was due to it trying to echo to `/etc/puppetlabs/code/hiera.yaml` which if installing by gem lacks all of that directory structure. Commenting it out didn't break anything so I'm not sure if it even needs to be there.

Possibly the `install_puppet_from_gem` helper should be enhanced to ensure post-installation that directories such as `/etc/puppetlabs/puppet`, `/etc/puppetlabs/code`, `/var/log/puppetlabs/puppet`, etc. should exist.